src/api/indicators/SimpleIndicator.js

```diff
@@ -0,0 +1,96 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2018, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define(['zepto', 'text!./res/indicator-template.html'],
+    function ($, indicatorTemplate) {
+        var DEFAULT_ICON_CLASS = 'icon-info';
+
+        function SimpleIndicator(openmct) {
+            this.openmct = openmct;
+            this.element = $(indicatorTemplate)[0];
+
+            this.textElement = this.element.querySelector('.indicator-text');
+
+            //Set defaults
+            this.text('New Indicator');
+            this.description('A simple indicator');
+            this.iconClass(DEFAULT_ICON_CLASS);
+            this.statusClass('');
+        }
+
+        SimpleIndicator.prototype.text = function (text) {
+            if (text !== undefined && text !== this.textValue) {
+                this.textValue = text;
+                this.textElement.innerText = text;
+
+                if (!text) {
+                    this.element.classList.add('hidden');
+                } else {
+                    this.element.classList.remove('hidden');
+                }
+            }
+
+            return this.textValue;
+        };
+
+        SimpleIndicator.prototype.description = function (description) {
+            if (description !== undefined && description !== this.descriptionValue) {
+                this.descriptionValue = description;
+                this.element.title = description;
+            }
+
+            return this.descriptionValue;
+        };
+
+        SimpleIndicator.prototype.iconClass = function (iconClass) {
```

Comment:
This is a nice middle road; I consider there to be two other options for this interface but I think this one is "just right".  

* isolate from style guide and simplify, i.e. predefine a set of icons/statuses and have them call `icon("clock"); status("connected");`, so they don't have to read style guide.
* tie more closely to style guide, have them set a single `cssClass` and then force them to understand that status classes can be combined with icon classes.

to be very clear: this is a comment, not a requested change.